Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+---@return boolean
+function PlusOnes:hasPlusOnes(name)
+    GL:debug("PlusOnes:hasPlusOnes");
 
-    return 0;
+    if (type(name) ~= "string") then
+        return false;
+    end
+
+    local normalizedName = self:normalizedName(name);
+
+    if (self.MaterializedData.DetailsByPlayerName[normalizedName]) then
+        return true;
+    end
+
+    return false;
+end]]--
+
+--- Get a player's plus one
+---
+---@param name string
+---@return void
+function PlusOnes:getPlusOnes(name)
+    GL:debug("PlusOnes:getPlusOnes");
+
+    if (type(name) ~= "string") then
+        return 0;
+    end
+
+    local normalizedName = self:normalizedName(name);
+
+    return GL:tableGet(self.MaterializedData or {}, "DetailsByPlayerName." .. normalizedName .. ".total", 0);
 end
 
---- Clear all plusOnes
+--- Set a player's plus one
 ---
+---@param name string
+---@param plusOne number
+---@param dontBroadcast boolean
 ---@return void
-function PlusOnes:clear()
-    GL:debug("PlusOnes:clear");
+function PlusOnes:setPlusOnes(name, plusOne, dontBroadcast)
+    GL:debug("PlusOnes:setPlusOnes");
+    if (type(name) ~= "string") then
+        return;
+    end
 
-    DB.PlusOnes = {};
+    dontBroadcast = GL:toboolean(dontBroadcast);
 
+    local normalizedName = self:normalizedName(name);
+
+    if (not self.MaterializedData.DetailsByPlayerName[normalizedName]) then
+        GL:tableSet(self.MaterializedData or {}, "DetailsByPlayerName." .. normalizedName .. ".total", plusOne);
+    else
+        self.MaterializedData.DetailsByPlayerName[normalizedName].total = plusOne;
+    end
+    DB:set("PlusOnes.Totals." .. normalizedName, plusOne);
+    DB:set("PlusOnes.MetaData.updatedAt", GetServerTime());
+
+    if (not dontBroadcast
+        and GL.Settings:get("PlusOnes.automaticallyShareData")
+    ) then
+        self:broadcastUpdate(normalizedName, plusOne);
+    end
     self:triggerChangeEvent();
 end
 
---- Get a player's PlusOnes value
+--- Delete an entry --- not sure where this would be needed since the player list is generated based on the group/raid members
+---
+---@param name string
+---@return void
+function PlusOnes:deletePlusOnes(name, dontBroadcast)
+    GL:debug("PlusOnes:deletePlusOnes");
+    if (type(name) ~= "string") then
+        return;
+    end
+
+    dontBroadcast = GL:toboolean(dontBroadcast);
+
+    local normalizedName = self:normalizedName(name);
+
+    if (not self.MaterializedData.DetailsByPlayerName[normalizedName]) then
+        return;
+    end
+
+    self.MaterializedData.DetailsByPlayerName[normalizedName] = nil;
+    DB.PlusOnes.Totals[normalizedName] = nil;
+    DB:set("PlusOnes.MetaData.updatedAt", GetServerTime());
+
+    if (not dontBroadcast
+        and GL.Settings:get("PlusOnes.automaticallyShareData")
+    ) then
+        self:broadcastUpdate(normalizedName, nil, nil, true);
+    end
+end
+
+--- Modify a player's plus one -- don't know if this is needed since numbers are increased or decreased and not set directly
```

Comment:
Same as above, should this stay in then?

Reply:
Same comment as above.